src/main/webapp/app/overview/course-conversations/layout/conversation-selection-sidebar/conversation-sidebar-section/conversation-sidebar-entry/conversation-sidebar-entry.component.ts

```diff
@@ -103,29 +115,46 @@ export class ConversationSidebarEntryComponent implements OnInit, OnDestroy {
                 takeUntil(this.ngUnsubscribe),
             )
             .subscribe(() => {
-                this.settingsChanged.emit();
+                this.settingsDidChange.emit();
             });
     }
+
     ngOnInit(): void {
-        this.hide$.pipe(debounceTime(100), distinctUntilChanged(), takeUntil(this.ngUnsubscribe)).subscribe((shouldHide) => {
-            this.conversationService.changeHiddenStatus(this.course.id!, this.conversation.id!, shouldHide).subscribe({
+        this.favorite$.pipe(debounceTime(100), distinctUntilChanged(), takeUntil(this.ngUnsubscribe)).subscribe((isFavorite) => {
```

Comment:
My recommendation would be to create a function called getFavoriteConversations() and put all this code in this function. This leads to a very clean ngOnInti() and a developer can easily see what the function does and don't have to read all the pipe and subscribe stuff. 
You could do this for hide and mute conversations, too.

Reply:
Good idea